sdk/core/core-https/test/pipeline.spec.ts

```diff
@@ -0,0 +1,332 @@
+    assert.strictEqual(policies.length, 2);
+    assert.strictEqual(testPolicy2, policies[0]);
+    assert.strictEqual(testPolicy, policies[1]);
+  });
+
+  it("getOrderedPolicies honors afterPolicies", function() {
+    const pipeline = HttpsPipeline.create();
+    const testPolicy: PipelinePolicy = {
+      sendRequest: (request, next) => next(request),
+      name: "test"
+    };
+    const testPolicy2: PipelinePolicy = {
+      sendRequest: (request, next) => next(request),
+      name: "test2"
+    };
+    pipeline.addPolicy(testPolicy, { afterPolicies: [testPolicy2.name] });
+    pipeline.addPolicy(testPolicy2);
+    const policies = pipeline.getOrderedPolicies();
+    assert.strictEqual(policies.length, 2);
+    assert.strictEqual(testPolicy2, policies[0]);
+    assert.strictEqual(testPolicy, policies[1]);
+  });
+
+  it("addPolicy throws on duplicate policy name", function() {
+    const pipeline = HttpsPipeline.create();
+    const testPolicy: PipelinePolicy = {
+      sendRequest: (request, next) => next(request),
+      name: "test"
+    };
+    const testPolicy2: PipelinePolicy = {
+      sendRequest: (request, next) => next(request),
+      name: "test"
+    };
+    pipeline.addPolicy(testPolicy);
+    pipeline.addPolicy(testPolicy2);
+    assert.throws(() => {
+      pipeline.getOrderedPolicies();
+    }, /Duplicate policy/);
+  });
+
+  it("getOrderedPolicies throws on circular reference", function() {
+    const pipeline = HttpsPipeline.create();
+    const testPolicy: PipelinePolicy = {
+      sendRequest: (request, next) => next(request),
+      name: "test"
+    };
+    const testPolicy2: PipelinePolicy = {
+      sendRequest: (request, next) => next(request),
+      name: "test2"
+    };
+    const testPolicy3: PipelinePolicy = {
+      sendRequest: (request, next) => next(request),
+      name: "test3"
+    };
+    pipeline.addPolicy(testPolicy, { afterPolicies: [testPolicy2.name] });
+    pipeline.addPolicy(testPolicy2, { afterPolicies: [testPolicy3.name] });
+    pipeline.addPolicy(testPolicy3, { afterPolicies: [testPolicy.name] });
+    assert.throws(() => {
+      pipeline.getOrderedPolicies();
+    }, /cycle/);
+  });
+
+  it("addPolicy orders within a correctly", function() {
+    const pipeline = HttpsPipeline.create();
+    const testPolicy: PipelinePolicy = {
+      sendRequest: (request, next) => next(request),
+      name: "test"
+    };
+    const testPolicy2: PipelinePolicy = {
+      sendRequest: (request, next) => next(request),
+      name: "test2"
+    };
+    const testPolicy3: PipelinePolicy = {
+      sendRequest: (request, next) => next(request),
+      name: "test3"
+    };
+    pipeline.addPolicy(testPolicy, { phase: "Retry" });
+    pipeline.addPolicy(testPolicy2, { afterPhase: "Retry" });
+    pipeline.addPolicy(testPolicy3, { phase: "Retry" });
+
+    const policies = pipeline.getOrderedPolicies();
+    assert.strictEqual(policies.length, 3);
+    assert.strictEqual(testPolicy, policies[0]);
```

Comment:
nit: actual should be the first argument.  `policies` here is the actual result.

Reply:
good catch! fixed this throughout.